src/System.Buffers/tests/ArrayPool/UnitTests.cs

```diff
@@ -211,5 +211,22 @@ public static void RentingAllBuffersAndCallingRentAgainWillAllocateBufferAndRetu
             Assert.NotNull(rented1);
             Assert.NotNull(rented2);
         }
+
+        [Fact]
+        public static void ReturningARentedBufferTwiceThrowsAnException()
+        {
+            using (ArrayPoolEventListener listener = new ArrayPoolEventListener())
```

Comment:
Using https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Diagnostics/Tracing/TestEventListener.cs, we should ideally have tests for all of the events, making sure they're firing when expecting, producing appropriate data, etc.